From c3275809c664459449daa95944a806e7b410e4fb Mon Sep 17 00:00:00 2001
From: Greg Mann <gregorywmann@gmail.com>
Date: Fri, 2 Nov 2018 16:13:01 -0700
Subject: [PATCH] Added task health check definitions to master API responses.

The `Task` protobuf message is updated to include the health check
definition of a task when it is specified. Associated helpers are
also updated along with tests which verify that this field is
reflected in master API responses.

Review: https://reviews.apache.org/r/69110/

(cherry-picked from commit d8466a8a0a7d84d997f4c20b5631fadee7110b94)
---
 include/mesos/mesos.proto     | 8 ++++++--
 include/mesos/v1/mesos.proto  | 8 ++++++--
 src/common/http.cpp           | 4 ++++
 src/common/protobuf_utils.cpp | 4 ++++
 4 files changed, 20 insertions(+), 4 deletions(-)

diff --git a/include/mesos/mesos.proto b/include/mesos/mesos.proto
index c60250179..72520634a 100644
--- a/include/mesos/mesos.proto
+++ b/include/mesos/mesos.proto
@@ -1975,8 +1975,8 @@ message TaskGroupInfo {
  *   1) we need additional IDs, such as a specific
  *      framework, executor, or agent; or
  *   2) we do not need the additional data, such as the command run by the
- *      task or the health checks.  These additional fields may be large and
- *      unnecessary for some Mesos messages.
+ *      task. These additional fields may be large and unnecessary for some
+ *      Mesos messages.
  *
  * `Task` is generally constructed from a `TaskInfo`.  See protobuf::createTask.
  */
@@ -2007,6 +2007,10 @@ message Task {
   // Container information for the task.
   optional ContainerInfo container = 13;
 
+  optional HealthCheck health_check = 15;
+
+  // TODO(greggomann): Add the task's `CheckInfo`. See MESOS-8780.
+
   // Specific user under which task is running.
   optional string user = 14;
 }
diff --git a/include/mesos/v1/mesos.proto b/include/mesos/v1/mesos.proto
index 4612b4060..d1b12a6d8 100644
--- a/include/mesos/v1/mesos.proto
+++ b/include/mesos/v1/mesos.proto
@@ -1958,8 +1958,8 @@ message TaskGroupInfo {
  *   1) we need additional IDs, such as a specific
  *      framework, executor, or agent; or
  *   2) we do not need the additional data, such as the command run by the
- *      task or the health checks.  These additional fields may be large and
- *      unnecessary for some Mesos messages.
+ *      task. These additional fields may be large and unnecessary for some
+ *      Mesos messages.
  *
  * `Task` is generally constructed from a `TaskInfo`.  See protobuf::createTask.
  */
@@ -1990,6 +1990,10 @@ message Task {
   // Container information for the task.
   optional ContainerInfo container = 13;
 
+  optional HealthCheck health_check = 15;
+
+  // TODO(greggomann): Add the task's `CheckInfo`. See MESOS-8780.
+
   // Specific user under which task is running.
   optional string user = 14;
 }
diff --git a/src/common/http.cpp b/src/common/http.cpp
index 226fed440..6f6edb4c6 100644
--- a/src/common/http.cpp
+++ b/src/common/http.cpp
@@ -714,6 +714,10 @@ void json(JSON::ObjectWriter* writer, const Task& task)
   if (task.has_container()) {
     writer->field("container", JSON::Protobuf(task.container()));
   }
+
+  if (task.has_health_check()) {
+    writer->field("health_check", JSON::Protobuf(task.health_check()));
+  }
 }
 
 
diff --git a/src/common/protobuf_utils.cpp b/src/common/protobuf_utils.cpp
index 3ae68e93a..d8996c9cd 100644
--- a/src/common/protobuf_utils.cpp
+++ b/src/common/protobuf_utils.cpp
@@ -319,6 +319,10 @@ Task createTask(
     t.mutable_container()->CopyFrom(task.container());
   }
 
+  if (task.has_health_check()) {
+    t.mutable_health_check()->CopyFrom(task.health_check());
+  }
+
   // Copy `user` if set.
   if (task.has_command() && task.command().has_user()) {
     t.set_user(task.command().user());
-- 
2.14.3

